pkg/scheduler/cache/event_handlers.go

```diff
@@ -793,3 +793,87 @@ func (sc *SchedulerCache) addPriorityClass(pc *v1beta1.PriorityClass) {
 
 	sc.PriorityClasses[pc.Name] = pc
 }
+
+func (sc *SchedulerCache) updateResourceQuota(quota *v1.ResourceQuota) {
+	collection, ok := sc.NamespaceCollection[quota.Namespace]
+	if !ok {
+		collection = kbapi.NewNamespaceCollection(quota.Namespace)
+		sc.NamespaceCollection[quota.Namespace] = collection
+	}
+
+	collection.Update(quota)
+}
+
+func (sc *SchedulerCache) deleteResourceQuota(quota *v1.ResourceQuota) {
+	collection, ok := sc.NamespaceCollection[quota.Namespace]
+	if !ok {
+		return
+	}
+
+	collection.Delete(quota)
+}
+
+// DeleteResourceQuota delete ResourceQuota from the scheduler cache
+func (sc *SchedulerCache) DeleteResourceQuota(obj interface{}) {
+	var r *v1.ResourceQuota
+	switch t := obj.(type) {
+	case *v1.ResourceQuota:
+		r = t
+	case cache.DeletedFinalStateUnknown:
+		var ok bool
+		r, ok = t.Obj.(*v1.ResourceQuota)
+		if !ok {
+			glog.Errorf("Cannot convert to *v1beta1.PriorityClass: %v", t.Obj)
+			return
+		}
+	default:
+		glog.Errorf("Cannot convert to *v1beta1.PriorityClass: %v", t)
+		return
+	}
+
+	sc.Mutex.Lock()
+	defer sc.Mutex.Unlock()
+
+	glog.V(3).Infof("Delete ResourceQuota <%s/%v> in cache", r.Namespace, r.Name)
+	sc.deleteResourceQuota(r)
+}
+
+// UpdateResourceQuota update ResourceQuota to scheduler cache
+func (sc *SchedulerCache) UpdateResourceQuota(oldObj, newObj interface{}) {
+	newR, ok := newObj.(*v1.ResourceQuota)
+	if !ok {
+		glog.Errorf("Cannot convert newObj to *v1beta1.PriorityClass: %v", newObj)
+		return
+	}
+
+	sc.Mutex.Lock()
+	defer sc.Mutex.Unlock()
+
+	glog.V(3).Infof("Update ResourceQuota <%s/%v> in cache, with spec: %v.", newR.Namespace, newR.Name, newR.Spec.Hard)
+	sc.updateResourceQuota(newR)
+}
+
+// AddResourceQuota add ResourceQuota to scheduler cache
+func (sc *SchedulerCache) AddResourceQuota(obj interface{}) {
```

Comment:
update namespace weight in `quota.status.used` for avoiding error `status unknown for quota: compute-resources`